loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/TransformersValidation.java

```diff
@@ -0,0 +1,246 @@
+        }
+    }
+
+    public static boolean checkTransformers(Network network, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+
+        try (ValidationWriter twtsWriter = ValidationUtils.createValidationWriter(network.getId(), config, writer, ValidationType.TWTS)) {
+            return checkTransformers(network, config, twtsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkTransformers(Network network, ValidationConfig config, ValidationWriter twtsWriter) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(twtsWriter);
+
+        LOGGER.info("Checking transformers of network {}", network.getId());
+        return network.getTwoWindingsTransformerStream()
+                      .filter(TransformersValidation::filterTwt)
+                      .sorted(Comparator.comparing(TwoWindingsTransformer::getId))
+                      .map(twt -> checkTransformer(twt, config, twtsWriter))
+                      .reduce(Boolean::logicalAnd)
+                      .orElse(true);
+    }
+
+    private static boolean filterTwt(TwoWindingsTransformer twt) {
+        return twt.getRatioTapChanger() != null && twt.getRatioTapChanger().isRegulating();
+    }
+
+    public static boolean checkTransformer(TwoWindingsTransformer twt, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(twt);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+
+        try (ValidationWriter twtsWriter = ValidationUtils.createValidationWriter(twt.getId(), config, writer, ValidationType.TWTS)) {
+            return checkTransformer(twt, config, twtsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkTransformer(TwoWindingsTransformer twt, ValidationConfig config, ValidationWriter twtsWriter) {
+        Objects.requireNonNull(twt);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(twtsWriter);
+
+        RatioTapChanger ratioTapChanger = twt.getRatioTapChanger();
+        int tapPosition = ratioTapChanger.getTapPosition();
+        int lowTapPosition = ratioTapChanger.getLowTapPosition();
+        int highTapPosition = ratioTapChanger.getHighTapPosition();
+        float rho = ratioTapChanger.getCurrentStep().getRho();
+        float rhoPreviousStep = tapPosition == lowTapPosition ? Float.NaN : ratioTapChanger.getStep(tapPosition - 1).getRho();
+        float rhoNextStep = tapPosition == highTapPosition ? Float.NaN : ratioTapChanger.getStep(tapPosition + 1).getRho();
+        float targetV = ratioTapChanger.getTargetV();
+        Side regulatedSide = twt.getTerminal1().equals(ratioTapChanger.getRegulationTerminal()) ? Side.ONE : Side.TWO;
```

Comment:
Could the regulation terminal be another terminal than the teminal1 or terminal2 of the transformer? You probably have to handle this case here